dowhy/causal_refuters/graph_refuter.py

```diff
@@ -1,69 +1,133 @@
-        else:
-            self.logger.error("Invalid conditional independence test")
+                    self.logger.error("Invalid conditional independence test for continuous data. Supported tests - partial_correlation")
+
+            elif a in discrete_columns and b in discrete_columns and all(node in discrete_columns for node in c):
+                # a, b and c are all discrete variables
+                if self._method_name_discrete is None or self._method_name_discrete == "conditional_mutual_information":
+                    self.conditional_mutual_information(x = a, y = b, z =c)
+                else:
+                    self.logger.error("Invalid conditional independence test for discrete data. Supported tests - conditional_mutual_information")
+
+            elif all(node in discrete_columns for node in c): 
+                # c is discrete and either a or b is continuous and the other is discrete
+                self.conditional_mutual_information(x = a, y= b, z = c)
+            elif all(node in continuous_columns for node in c) and ((a in continuous_columns and b in binary_columns) or (a in binary_columns and b in continuous_columns) or (a in binary_columns and b in binary_columns) ): 
+                # c is continuous and 
+                #   1. either a and b is continuous and the other is binary
+                #   2. both and b are binary
+                self.partial_correlation(x = a, y= b, z = c)
+            elif all(node in continuous_columns for node in c) and ((a in continuous_columns and b not in binary_columns) or (b in continuous_columns and a not in binary_columns)):
+                # c is continuous and either of a and b is continuous and the other is discrete-non binary
+                print("The following setting not supported")
+            elif a in continuous_columns and b in continuous_columns and all(node in continuous_columns  or node in binary_columns for node in c):
+                # a and b are continuous and c has set of continuous and binary variables
+                self.partial_correlation(x = a, y= b, z = c) 
+            elif ((a in continuous_columns and b in binary_columns) or (a in binary_columns and b in continuous_columns)) and all(node in continuous_columns  or node in binary_columns for node in c):
+                # Either of a and b is continuous and other one is binary and c has continuous and binary variables
+                self.partial_correlation(x = a, y= b, z = c) 
+            elif a in discrete_columns and b in discrete_columns:
+                # a and b are discrete and c is a mixture of discrete and continuous variables. We discretize c and calculate conditional mutual information
+                self.conditional_mutual_information(x = a, y= b, z = c)
+            else:
+                print("The following setting not supported")
```

Comment:
instead of "following", can you output the specific setting. E.g., the setting with a as discrete, b as non-discrete, etc.

Reply:
also you should return the function immediately, with a refute object that has Undecided or None as the test result. 
actually, None is better.